samples/hello_world/src/main.c

```diff
@@ -5,8 +5,23 @@
  */
 
 #include <zephyr/kernel.h>
+#include <zephyr/drivers/pcie/pcie.h>
 
 void main(void)
 {
+	int dev_count;
+
 	printk("Hello World! %s\n", CONFIG_BOARD);
+
+	STRUCT_SECTION_COUNT(pcie_dev, &dev_count);
+	printk("%d\n", dev_count);
+
+	STRUCT_SECTION_FOREACH(pcie_dev, dev) {
```

Comment:
Can some of this be added to tests (if there is appropriate place)

Reply:
I didn't intend to push this commit. Removed now.